webapp/src/js/services/format-data-record.js

```diff
@@ -180,13 +181,29 @@ angular.module('inboxServices').factory('FormatDataRecord',
               });
     };
 
-    var formatDate = function(settings, date) {
+    var isToday = function(momento) {
+      var startOfDay = moment().startOf('day');
+      var duration = moment.duration(startOfDay.diff(momento));
+      return duration.asHours < 24;
+    };
+
+    var formatDateField = function(settings, date, field) {
       if (!date) {
-          return;
+        return;
       }
+      var dailyPrecisionFields = [
+        'child_birth_date',
+        'birth_date'
+      ];
+
       var m = moment(date);
+      var fromNow = m.fromNow();
+      if (_.contains(dailyPrecisionFields, field) && isToday(m)) {
+        fromNow = FormatDate.relative(m, { withoutTime: true });
+      }
+
       return m.format(settings.date_format) +
-        ' (' + m.fromNow() + ')';
+        ' (' + fromNow + ')';
```

Comment:
I still think there is way too much code here.

What happens if you replace _everything_ here with this:

```js
var formatDateField = function(date, field) {
  if (!date) {
    return;
  }
  var formatted;
  var relative;
  if (_.contains([ 'child_birth_date', 'birth_date' ], field)) {
    formatted = FormatDate.date(date);
    relative = FormatDate.relative(date, { withoutTime: true });
  } else {
    formatted = FormatDate.datetime(date);
    relative = FormatDate.relative(date);
  }
  return formatted + '(' + relative + ')';
};
```

This is consistent with how dates are shown in other areas, and uses well tested routines rather than reimplementing them.

NB: I've dropped the `settings` parameter which is no longer needed in this function.
NB2: I haven't tested this at all.